### ISSUE TITLE
Problem for Ubuntu 20.04 with AMD Ryzen CPU when upgrading PCL 1.10 to Higher version

### ISSUE THREAD

**Issue by Seekerzero** [koide3/iridescence#142]
Hi Koide,
I was upgrading my PCL(1.10.1) to adapt your small_gicp library. And I discovered a weird problem with iridescence. There is a problem when launching the `viewer::instance()`, which causes segmentation faults and crashes. I entered the debug mode and figured out it was the shared_ptr writing in AVXINTRIN.h files. I also looked into the memory stack; mostly, the features that trigger the faults are from the ` shader setting` and `camera control`, and maybe more, but I haven't looked into it deeply.

I eventually solved the issue by turning on the `BUILD_WITH_MARCH_NATIVE` in the cmakelist of iridescence to solve the problem.

It is kinda weird that it didn't happen before on my end, only after switching to the PCL version, and also not really intuitive. Do you have any idea about this? 

Hopefully, this could be helpful for those meet the same issue.

**Comment by koide3** [comment]
Eigen requires all involved libraries to be built with the same `march=native` option, and if this is violated it causes segfault. The packages distributed via `apt` are built without `march=native`, and maybe your locally built PCL was built with it. I generally recommend keeping `march=native` disabled because it is often tough to make this option enabled for all libraries.

**Comment by Seekerzero** [comment]
I will try to look into my eigen build, thanks for your info.

**Comment by Seekerzero** [comment]
So I tried to turn the `march=native build` to OFF on PCL 1.11.1 and it worked and solved the previous issue. I also tried PCL 1.14.1 and it didn't work (maybe they changed some settings again). But I think it solved my issue and thanks for your excellent hint!

Some info that I found from online that might be helpful:[turn SSE OFF on PCL building](https://stackoverflow.com/questions/77540319/how-to-change-march-native-compile-flag-added-by-pcl-in-cmake). The `-DPCL_ENABLE_MARCHNATIVE=False` is not necessary as it was not recognized in the cmake configuration for PCL 1.11.1 .

Also I found something to tweak on the cmakelist, I've already created a pull request. #143 

Thanks again!
